pkg/client/unversioned/deployment.go

```diff
@@ -33,6 +33,7 @@ type DeploymentInterface interface {
 	Get(name string) (*extensions.Deployment, error)
 	Delete(name string, options *api.DeleteOptions) error
 	Create(*extensions.Deployment) (*extensions.Deployment, error)
+	Patch(*extensions.Deployment, []byte) (*extensions.Deployment, error)
```

Comment:
This is already supported by the clientset, see [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/clientset_generated/release_1_4/typed/extensions/v1beta1/deployment.go#L41). We plan to deprecate the unversioned client so we didn't make the change here. That being said, it's good to update the unversioned package. Could you keep the interface same as that of clientset? It will make conversion to clientset a little easier.